src/Cli/Commands/ExportOptions.cs

```diff
@@ -1,27 +1,63 @@
         {
             GraphQL = graphql;
             OutputDirectory = outputDirectory;
-            GraphQLSchemaFile = graphqlSchemaFile ?? "schema.graphql";
+            GraphQLSchemaFile = graphqlSchemaFile ?? "schema.gql";
+
+            Generate = generate ?? false;
+            SamplingMode = samplingMode ?? SamplingModes.TopNSampler.ToString();
+            NumberOfRecords = numberOfRecords;
+            PartitionKeyPath = partitionKeyPath;
+            MaxDays = maxDays;
+            GroupCount = groupCount;
         }
 
         [Option("graphql", HelpText = "Export GraphQL schema")]
         public bool GraphQL { get; }
 
-        [Option('o', "output", HelpText = "Directory to save to", Required = true)]
+        [Option('o', "output", HelpText = "Specify the directory where the schema file will be saved. This option is required.", Required = true)]
         public string OutputDirectory { get; }
 
-        [Option('g', "graphql-schema-file", HelpText = "The GraphQL schema file name (default schema.graphql)")]
+        [Option('g', "graphql-schema-file", HelpText = "Specify the filename for the exported GraphQL schema. Default is 'schema.gql'.")]
         public string GraphQLSchemaFile { get; }
+
+        [Option("generate", HelpText = "Generates a schema file from the specified CosmosDB database.")]
+        public bool Generate { get; }
+
+        [Option('m', "sampling-mode", HelpText = "Specifies the sampling mode to use. Available modes include: " +
+                                                 "- TopNSampler: It retrieves a specified number of recent records from a Cosmos DB container, optionally filtering by a maximum number of days." +
+                                                 "- PartitionBasedSampler: It retrieves a specified number of records from a Cosmos DB container by fetching records from each partition based on a specified partition key.The sampling is configurable by the number of records per partition and the time range considered." +
+                                                 "- TimeBasedSampler:.It retrieves a specified number of records by dividing the container data, time range into subranges and retrieving the top N records from each subrange based on a specified configuration")]
+        public string SamplingMode { get; } = SamplingModes.TopNSampler.ToString();
+
+        [Option('n', "sampling-count", HelpText = "Specify the total number of samples to be retrieved. For different sampling modes:\n" +
+                                                  "- TopNSampler: Total number of records to select.\n" +
+                                                  "- PartitionBasedSampler: Number of records to retrieve from each partition.\n" +
+                                                  "- TimeBasedSampler: Number of records to retrieve per time range group.")]
```

Comment:
I am not sure customers would be exposed to this , because we don't expose physical partition to customers . can't we abstract and have a default value?

Reply:
Here, we are talking about logical partition.